test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/testproviders/PersistenceTestProvider.java

```diff
@@ -85,6 +90,134 @@ public void testControllerPersistence(WorkbenchKieServerScenario deploymentScena
+            SoftAssertions.assertSoftly(softly -> {
+                softly.assertAlso(isUserAuthorizedInWorkbench(workbenchUrl, oldUsername, oldPassword));
+                softly.assertAlso(isUserAuthorizedInKieServer(kieServerUrl, oldUsername, oldPassword));
+            });
+
+            logger.info("Change username to {} and password to {} in credential secret", NEW_USERNAME, NEW_PASSWORD);
+            deploymentScenario.changeUsernameAndPassword(NEW_USERNAME, NEW_PASSWORD);
+            //workbenchDeployment.changeUsernameAndPassword(NEW_USERNAME, NEW_PASSWORD);
+
+            SoftAssertions.assertSoftly(softly -> {
+                softly.assertAlso(isUserUnauthorizedInWorkbench(workbenchUrl, oldUsername, oldPassword));
+                softly.assertAlso(isUserUnauthorizedInWorkbench(workbenchUrl, oldUsername, NEW_PASSWORD));
+                softly.assertAlso(isUserUnauthorizedInWorkbench(workbenchUrl, NEW_USERNAME, oldPassword));
+
+                softly.assertAlso(isUserAuthorizedInWorkbench(workbenchUrl, NEW_USERNAME, NEW_PASSWORD));
+
+                softly.assertAlso(isUserUnauthorizedInKieServer(kieServerUrl, oldUsername, oldPassword));
+                softly.assertAlso(isUserUnauthorizedInKieServer(kieServerUrl, oldUsername, NEW_PASSWORD));
+                softly.assertAlso(isUserUnauthorizedInKieServer(kieServerUrl, NEW_USERNAME, oldPassword));
+
+                softly.assertAlso(isUserAuthorizedInKieServer(kieServerUrl, NEW_USERNAME, NEW_PASSWORD));
+            });
+        } finally {
+            deploymentScenario.changeUsernameAndPassword(oldUsername, oldPassword);
+            kieControllerClient.deleteContainerSpec(serverInfo.getServerId(), containerId);
+        }
+    }
+
+    private SoftAssertions isUserAuthorizedInWorkbench(String url, String username, String password) {
+        KieServerControllerClient kieControllerClient = KieServerControllerClientProvider.getKieServerControllerClient(url, username, password);
+        SoftAssertions softly = new SoftAssertions();
+        softly.assertThat(kieControllerClient.listServerTemplates().getServerTemplates())
+              .as("List Server Templates after login with username {} and password {}", username, password)
+              .hasSize(1);
+        return softly;
+    }
+
+    private SoftAssertions isUserUnauthorizedInWorkbench(String url, String username, String password) {
+        KieServerControllerClient kieControllerClient = KieServerControllerClientProvider.getKieServerControllerClient(url, username, password);
+        SoftAssertions softly = new SoftAssertions();
+        softly.assertThatThrownBy(() -> {
+            kieControllerClient.listServerTemplates().getServerTemplates();
+        }).as("Exception after login with username {} and password {}", username, password)
+              .isInstanceOf(IndexOutOfBoundsException.class) //TODO
+              .hasMessageContaining("Index: 2, Size: 2"); //TODO
```

Comment:
Why this TODO? What is pending?

Reply:
Wrong exception instance was there checked. fixed